### PR TITLE
Preliminary RustDesk inventory support

### PR DESCRIPTION
**Pull request by paspo**
A simple inventory support for [rustdesk](https://rustdesk.com/), an open-source remote management tool.
I took a lot of inspiration from the AnyDesk counterpart.

### CODE REVIEW

File: lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+sub _get_rustdesk_config {
+    my @configs = ();
+    if (OSNAME eq 'MSWin32') {
+        if (has_folder('C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk')) {
+            push @configs, 'C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml';
+        }
+    } else {
+        push @configs, '/root/.config/rustdesk/RustDesk.toml';
+    }
+    return grep { has_file($_) } @configs;
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm]: Here you should better just return the config file path expected on the current OS.
```suggestion
    return OSNAME eq 'MSWin32' ?
        'C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml' :
        '/root/.config/rustdesk/RustDesk.toml';
```

File: lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+sub isEnabled {
+    return _get_rustdesk_config() ? 1 : 0;
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm]: With the upper change, you should modify here:
```suggestion
    return has_file(_get_rustdesk_config());
```

File: lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                section => 'REMOTE_MGMT',
+                entry   => {
+                    ID   => $RustDeskID,
+                    TYPE => 'rustdesk'
+                }
+            );
+        } else {
+            $logger->debug('RustDesk ID not found in '.$conf) if $logger;
+        }
+    }
```
Comment (g-bougard) [inline lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/RustDesk.pm]: Here you don't need a loop as you expect only one value from _get_rustdesk_config():
```suggestion
    my $conf = _get_rustdesk_config();
    my $RustDeskID = getFirstMatch(
        file    => $conf,
        logger  => $logger,
        pattern => qr/^id\s*=\s*'(.*)'$/
    );

    if (defined($RustDeskID)) {
        $logger->debug('Found RustDesk ID : ' . $RustDeskID) if $logger;

        $inventory->addEntry(
            section => 'REMOTE_MGMT',
            entry   => {
                ID   => $RustDeskID,
                TYPE => 'rustdesk'
            }
        );
    } else {
        $logger->debug('RustDesk ID not found in '.$conf) if $logger;
    }
```

**Review by g-bougard — CHANGES_REQUESTED** [review]
Hi @paspo 

thank you for your submission.

I see a little changed needed as `_get_rustdesk_config()` will always return the expected path of the config for the current OS.
The big proposed patch in `doInventory()` is just to remove the loop.

Supporting a list of configuration files make sense only if we really provide a list in some case.

**Comment by paspo** [comment]
@g-bougard thanks a lot for the (blazing fast) review!

I think we still need https://github.com/glpi-project/glpi/pull/12406

**Comment by g-bougard** [comment]
Hi @paspo 
yes, glpi-project/glpi#12406 is still needed. I'll keep this PR opened until glpi one is merged.

**Review by g-bougard — APPROVED** [review]

**Comment by paspo** [comment]
> I'll keep this PR opened until glpi one is merged.

the other PR got accepted.

**Comment by billgatos** [comment]
Hi all,

Using Nightly latest as today, i get in my local json file after inventory the info for rustdesk id

<REMOTE_MGMT>
<ID>99999999</ID>
<TYPE>anydesk</TYPE>
</REMOTE_MGMT>
<REMOTE_MGMT>
<ID>99999999</ID>
<TYPE>rustdesk</TYPE>
</REMOTE_MGMT>

But that info isn't loaded to glpi ?????

any thoughts on this .

Best regards.

**Comment by billgatos** [comment]
hi all,

The following recepie works:
GLPI 10.0.5 and agent 1.5 nightly gets the rustdesk id to GLPI asset.

**Pull request merged**